book/testing_code.md

```diff
@@ -535,15 +536,15 @@ To ensure that the model generalises well to unseen data, techniques like k-fold
 
 ### Stress Testing
 
-Stress testing evaluates how the model performs under extreme conditions or with noisy data. This helps identify the model's robustness and ability to handle unexpected inputs. Stress testing involves introducing variations or noise into the input data and observing how the model's predictions are affected. This type of testing is useful for understanding the model's limits and ensuring it can handle real-world challenges.
+Stress testing evaluates how the model performs under extreme conditions or with noisy data. This helps identify the model's robustness and ability to handle unexpected inputs. Stress testing involves introducing variations or noise into the input data and observing how the model's outputs are affected. This type of testing is useful for understanding the model's limits and ensuring it can handle real-world challenges.
 
 ### Sensitivity Analysis
 
-Sensitivity analysis tests how sensitive the model's outputs are to changes in input data or parameters. This analysis helps understand the model's behaviour and identify potential weaknesses. Sensitivity analysis involves systematically varying the input data or model parameters and measuring the impact on the model's predictions. This helps in identifying critical factors that influence the model's performance and making necessary adjustments.
+Sensitivity analysis tests how sensitive the model's outputs are to changes in input data or parameters. This analysis helps understand the model's behaviour and identify potential weaknesses. Sensitivity analysis involves systematically varying the input data or model parameters and measuring the impact on the model's outputs. This helps in identifying critical factors that influence the model's performance and making necessary adjustments.
 
 ### Model Interpretability
 
-Implementing methods to make the model's predictions interpretable is essential for building trust with stakeholders. Techniques like SHAP (SHapley Additive exPlanations) values or LIME (Local Interpretable Model-agnostic Explanations) can help explain the model's decisions. These methods provide insights into how different features contribute to the model's predictions, making it easier for analysts and stakeholders to understand and trust the model's outputs.
+Implementing methods to make the model's outputs interpretable is essential for building trust with stakeholders. Techniques like SHAP (SHapley Additive exPlanations) values or LIME (Local Interpretable Model-agnostic Explanations) can help explain the model's decisions. These methods provide insights into how different features contribute to the model's outputs, making it easier for analysts and stakeholders to understand and trust the model's outputs.
 
 ### Model Optimisation
```

Comment:
Again, start with the general. I don't know what grid search and hyperparameter tuning are. You could rework this para to make it clearer:

Optimisation is used to adjust adjusting the model's parameters to achieve the best overall performance. Continuous optimisation ensures that the model remains effective and efficient over time as inputs change. There are lots of techniques available to optimise performance. Most are designed to help find the best parameters for the model to enhance its accuracy and efficiency. 

Examples of optimisation techniques in the context of machine learning include grid search and parameter tuning. [Grid search involves systematically searching through a predefined set of hyperparameters, while hyperparameter tuning adjusts the model's parameters to achieve the best possible performance. ] NOTE: These are very jargon heavy.... most people won't know what they mean....

The Use fixtures section seems to apply more widely than just to modelling. I think this para should be moved to the wider discussion of testing. Again, beware of introducing jargon terms (fixtures, parameterised tests) that people may not understand without resources to support them.